### ISSUE TITLE
Unexpected token = in class constructor shorthand

### ISSUE THREAD

**Issue by Redsandro** [eslint/eslint#12586]
### Environment Info:

**Node version:** v12.13.0
**npm version:** v6.12.0
**Local ESLint version:** Not found
**Global ESLint version:** v6.6.0 (Currently used)

<details>
<summary>Configuration</summary>

```js
{
    "parserOptions": {
        "ecmaVersion": 2018,
        "sourceType": "module"
    },
    "env": {
        "node": true,
        "es6": true
    },
    "extends": "eslint:recommended"
}
```

</details>

```js
module.exports = class NotAcceptableError extends Error {
    status = 406
}
```

```bash
$ eslint test.js 

/tmp/estest/test.js
  2:12  error  Parsing error: Unexpected token =

✖ 1 problem (1 error, 0 warnings)
```

**What did you expect to happen?**

Lints without error

**What actually happened? Please include the actual, raw output from ESLint.**
> Unexpected token =

**Comment by g-plane** [comment]
Thanks for your report.

ESLint only supports Stage 4 features. To use class fields, you can use `babel-eslint` as parser.

**Comment by Redsandro** [comment]
I had to google that. Good to know, thank you.

https://2ality.com/2015/11/tc39-process.html
